### PR TITLE
SWProxy.py structural cleanups + Readability improvements

### PR DESCRIPTION
**Pull request by azrethos**
I moved the python requirements into a standard requirements.txt

Also helped make the SWProxy.py code a bit more structured and readable, let me know what you think. I am a big fan of abstracting away complex conditions into lambdas so the implementation code can look much cleaner.

Also added an optional debug flag to get more verbose logging. I'm for converting the args options into argparse. Also, do you want me to move the 3 main SWParser blocks into plugins? (visit friend, runes, etc)

### CODE REVIEW

**Comment by kakaroto** [comment]
Thanks, this is great. I can clearly see how much more experience you have with python than me, lol. Either way, I love everything that respects the "Code Is Art"(TM) motto. 
As for the code looking cleaner with lambdas and all that, I'm not sure about that, I guess only if you are used to it, for me it makes it hard to follow what exactly it does.
I think moving things to use argparse is a great idea. I quickly hacked a --no-gui option (like you did for debug) in the gui branch, but I wanted to move to using argparse because it would be so much less confusing.
Either things I was thinking of :
* I don't like having SWParser.py and SWParser/ module with the same name.
* I don't like that for each release, I have to change the VERSION variable in both SWParser.py and SWProxy.py
* I think it's time to change the name of the app into SWProxy since it evolved much more into a proxy than a pcap parser.
* With the GUI being built (and argparse being added), we can completely remove SWParser.py and have a single entry point handle both proxy and pcap methods (gui as a menu option, and cli as a cli arg).
* I need to add PyQt4 to the requirements.txt in the gui branch
* We need to have a much better README file that would be 100 times less confusing than it is right now
* README file needs an intro on using/developing plugins
* You need to add your name to the AUTHORS file, so it goes in with the merge.

**Comment by azrethos** [comment]
I'll make these changes;
